### PR TITLE
Add proposed api to support auto closing pairs on langauge configuration

### PR DESCRIPTION
**Pull request by colombod**
This pr provides support for auto closing pairs on language configuration

### CODE REVIEW

**Comment by colombod** [comment]
@jrieken, @alexdima this is the pr to address #173738

**Review by alexdima — DISMISSED** [review]
This is great! Thank you!

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts [hunk 1]
```diff
@@ -2542,7 +2542,9 @@ export class ExtHostLanguageFeatures implements extHostProtocol.ExtHostLanguageF
 			onEnterRules: configuration.onEnterRules ? ExtHostLanguageFeatures._serializeOnEnterRules(configuration.onEnterRules) : undefined,
 			__electricCharacterSupport: configuration.__electricCharacterSupport,
 			__characterPairSupport: configuration.__characterPairSupport,
+			autoClosingPairs: configuration.autoClosingPairs
```
Comment (alexr00) [inline src/vs/workbench/api/common/extHostLanguageFeatures.ts:2545]: @alexdima, should there be a `checkProposedApiEnabled` somewhere here?

Comment (alexdima) [reply src/vs/workbench/api/common/extHostLanguageFeatures.ts:2549]: Good catch! I've updated the PR.

**Review by alexdima — APPROVED** [review]

**Review by ulugbekna — APPROVED** [review]

**Pull request merged**